File: backend/app/models/candidate.py
from typing import Dict, List, Optional

import pymongo
from beanie import Document
from pydantic import Field


class Candidate(Document):
    hh_resume_id: str
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    age: Optional[int] = None
    gender: Optional[str] = None
    area: Optional[str] = None
    title: Optional[str] = None
    salary_amount: Optional[int] = None
    salary_currency: Optional[str] = None
    skills: List[str] = []
    photo_url: Optional[str] = None
    resume_url: Optional[str] = None
    is_vectorized: bool = False
    is_saved: bool = True  # False for hh matches not yet confirmed into the candidate database
    raw_resume_json: Dict = Field(default_factory=dict)

    class Settings:
        name = "candidates"
        indexes = [
            pymongo.IndexModel(
                [("hh_resume_id", pymongo.ASCENDING)],
                unique=True,
            )
        ]

File: frontend/types/candidate.ts
export interface Candidate {
  id: string;
  vacancy_id: string | null;
  hh_resume_id: string;
  first_name: string | null;
  last_name: string | null;
  age: number | null;
  gender: string | null;
  area: string | null;
  title: string | null;
  salary_amount: number | null;
  salary_currency: string | null;
  skills: string[];
  photo_url: string | null;
  resume_url: string | null;
  relevance_score: number | null;
  vector_score: number | null;
  llm_score: number | null;
  is_vectorized: boolean;
  matched_at: string;
  // Source & contact (from raw_resume_json)
  source: string | null;       // "xlsx" | "file" | "hh" | null
  phone: string | null;
  comment: string | null;
  score_reasoning: string | null;
  score_criteria: ScoreCriterion[] | null;
}

export interface ScoreCriterion {
  name: string;
  weight: number;
  score: number;
  comment: string;
}

export interface CandidateDetail extends Candidate {
  raw_resume_json: Record<string, unknown>;
}

